amundsen_application/static/js/pages/TableDetailPage/index.tsx

```diff
@@ -105,13 +126,26 @@ const ErrorMessage = () => {
   );
 };
 
+export type SortDirection = 'asc' | 'desc';
+export type SortCriteria = { key: string; direction: SortDirection };
+
+export interface StateProps {
+  sortedBy: SortCriteria;
+}
+
 export class TableDetail extends React.Component<
-  TableDetailProps & RouteComponentProps<any>
+  TableDetailProps & RouteComponentProps<any>,
+  StateProps
 > {
   private key: string;
 
   private didComponentMount: boolean = false;
 
+  // @ts-ignore
+  state = {
```

Comment:
What issue does this cause that we have to ignore it? Does the fact that we're not initializing in a `constructor` method have any effect?

Reply:
Not sure!
I don't think so, usually this method is preferred.

Reply:
Actually, it doesn't fail anymore. Updating it!